src/ape/contracts/base.py

```diff
@@ -126,6 +126,48 @@ def __call__(self, *args, **kwargs) -> Any:
             address=self.contract.address,
         )(*args, **kwargs)
 
+    def as_transaction(self, *args, **kwargs):
+        """
+        Convert the call to a transaction. This is useful for checking coverage
+        or checking gas costs.
+
+        Args:
+            *args: The contract method invocation arguments.
+            **kwargs: Transaction kwargs, such as value or
+              sender.
+
+        Returns:
+            :class:`~ape.api.transactions.TransactionAPI`
+        """
+        args = self._convert_tuple(args)
+        selected_abi = _select_method_abi(self.abis, args)
+
+        contract_transaction = ContractTransaction(  # type: ignore
+            abi=selected_abi,
+            address=self.contract.address,
+        )
+
+        transaction = contract_transaction.serialize_transaction(*args, **kwargs)
+        self.provider.prepare_transaction(transaction)
+        return transaction
+
+    def estimate_gas_cost(self, *args, **kwargs) -> int:
+        """
+        Get the estimated gas cost (according to the provider) for the
+        contract method call (as if it were a transaction).
+
+        Args:
+            *args: The contract method invocation arguments.
+            **kwargs: Transaction kwargs, such as value or
+              sender.
+
+        Returns:
+            int: The estimated cost of gas to execute the transaction
+            reported in the fee-currency's smallest unit, e.g. Wei.
+        """
+        txn = self.as_transaction(*args, **kwargs)
```

Comment:
In my hot-patch, I added a check right after that line:

```python
if txn.max_fee:
    return txn.max_fee
```

It may not be correct, but it was required because of those lines in `ape-starknet`: https://github.com/ApeWorX/ape-starknet/blob/6a9b090a89c71a2c1517e9502dad252452705da9/ape_starknet/provider.py#L317-L318

So after `.as_transaction()` was called, calling again `.estimate_gas_cost()` would end on a HTTP 500 error. Maybe that should be dealt at the `ape-starknet` level, I'm just rising the question :)